app/Http/Requests/StoreImageAnnotationLabel.php

```diff
@@ -30,7 +30,14 @@ class StoreImageAnnotationLabel extends FormRequest
     public function authorize()
     {
         $this->annotation = ImageAnnotation::findOrFail($this->route('id'));
-        $this->label = Label::findOrFail($this->input('label_id'));
+
+        $labelId = $this->input('label_id');
+
+        if (!is_numeric($labelId) || intval($labelId) != $labelId) {
+            return false;
+        }
+
+        $this->label = Label::findOrFail($labelId);
```

Comment:
Returning `false` here would imply that the user is unauthorized to make the request. I'd rather return a validation error. This could be done like this:

```suggestion
        $this->validate(['label_id' => 'integer']);
        $this->label = Label::findOrFail($this->input('label_id'));
```

That should be enough for the authorize method. The more extensive validation (required, exists) is done later.